prow/pod-utils/gcs/upload.go

```diff
@@ -44,24 +42,10 @@ type destToWriter func(dest string) dataWriter
 func Upload(bucket, gcsCredentialsFile, s3CredentialsFile string, uploadTargets map[string]UploadFunc) error {
 	parsedBucket, err := url.Parse(bucket)
 	if err != nil {
-		return fmt.Errorf("cannot parse bucket name %s: %v", bucket, err)
+		return fmt.Errorf("cannot parse bucket name %s: %w", bucket, err)
 	}
-	if parsedBucket.Scheme == "" || parsedBucket.Scheme == "gs" {
-		if parsedBucket.Scheme == "gs" {
-			bucket = strings.Trim(bucket, "gs://")
-		}
-		var options []option.ClientOption
-		if gcsCredentialsFile != "" {
-			options = append(options, option.WithCredentialsFile(gcsCredentialsFile))
-		}
-		gcsClient, err := storage.NewClient(context.Background(), options...)
-		if err != nil {
-			return fmt.Errorf("create client: %w", err)
-		}
-		dtw := func(dest string) dataWriter {
-			return gcsObjectWriter{gcsClient.Bucket(bucket).Object(dest).NewWriter(context.Background())}
-		}
-		return upload(dtw, uploadTargets)
+	if parsedBucket.Scheme == "" {
+		parsedBucket.Scheme = "gs"
```

Comment:
gcsObjectWriter struct should be unused now

Reply:
removed it